src/web/services/course.service.ts

```diff
@@ -25,6 +25,17 @@ export class CourseService {
     return this.httpRequestService.get('/course', paramMap);
   }
 
+  /**
+   * Gets all course data for an instructor/student by calling API.
+   */
+  getAllCourses(entityType: string, courseStatus: string): Observable<Courses> {
```

Comment:
Can change or not.

I prefer to hide the entityType in the service. A string type of entityType gives me a sense of nondeterministic :P.

Maybe just `getAllCoursesAsStudent`, `getAllCoursesAsInstructor`.

Reply:
Actually, those methods already exist in the service. So please change them :P

Reply:
Updated accordingly!